assets/tests/TaskSchemaValidationService.js

```diff
@@ -0,0 +1,225 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for validating data put into a task schema
+ * within a question.
+ */
+
+tie.factory('TaskSchemaValidationService', ['SYSTEM_CODE',
+  function(SYSTEM_CODE) {
+    var AUXILIARY_CODE_CLASS_NAME = 'AuxiliaryCode.';
+    var SYSTEM_CODE_CLASS_NAME = 'System.';
+    // TODO(eyurko): Update this once we support nonlinear runtimes.
+    var ALLOWED_RUNTIMES = ['linear'];
+    var _checkIfFunctionExistsInCode = function(functionName, code) {
```

Comment:
Consider adding TODOs to harden these two checks -- the first one for being at start of line (instead of an inner function); the second for actually making sure it exists in the class. The second one is quite weak, actually, since if the function is defined in the wrong class (i.e. with the wrong class name) then it won't catch it.